src/components/List/ListItem.tsx

```diff
@@ -14,7 +16,10 @@ export default function ListItem({
   onClick: (ref: React.MutableRefObject<unknown>, person: Person) => void;
 }) {
   const clickRef = useRef(null);
-  const { city, province } = getCityProvince(person.city);
+  const { getCity, getProvince } = useData();
```

Comment:
imho best would be to leverage getServerSideProps instead of using a data provider so that you can enable SSR.